File: trai-error/trai-error.java
import java.util.*;
import java.lang.Math;

public class logic{
	public static int count;
	public static void heron(int[] sd, int s){
		if(Math.sqrt( s * (s-sd[0]) * (s-sd[1]) * (s-sd[2]) ) == Math.floor(Math.sqrt( s * (s-sd[0]) * (s-sd[1]) * (s-sd[2]) ))){
			count+=1;
			// System.out.println(sd[0] + " " + sd[1] + " " + sd[2]);
		}
	}

	public static void sidesFromPerim(int p){
		int[] sd = new int[3];
		for(sd[0] = 1; sd[0] <= p / 2; sd[0]++){
			for(sd[1] = 1; sd[1] + sd[0] < p; sd[1]++){
				sd[2] = p - (sd[0] + sd[1]);
				if ( sd[0] + sd[1] <= sd[2] || sd[0] + sd[2] <= sd[1] || sd[1] + sd[2] <= sd[0] )
					continue;
				if (sd[0] <= sd[1] && sd[1] <= sd[2])
					heron(sd, p/2);
			}
		}
	}

	public static void main(String[] Args){
		Scanner sc = new Scanner(System.in);
		int t;
		t = sc.nextInt();
		while(t-->0){
			int p;
			p = sc.nextInt();
			count = 0;
			sidesFromPerim(p);
			System.out.println(count);
		}
	}
}

File: trai-error/trai-error.cpp
#include <bits/stdc++.h>
 
using namespace std;

int count = 0;

void heron(int *sd, double s){
	double d = s*(s - (1.0)**sd)*(s - (1.0)**(sd + 1))*(s - (1.0)**(sd + 2));
	double d1 = sqrt(d);
	long long d2 = (long long)sqrt(d);
	if(d1 == d2){
		::count+=1;
		}
}

void sidesFromPerim(int p){
	int s[3];
	for ( *s = 1; *s <= p / 3 ; (*s)++ ){
		*(s + 1) = floor( (p - *s * 2) / 2 + 1 );
			if ( *(s + 1) < *s ) *(s + 1) = *s;
			*(s + 2) = p - *s - *(s + 1);
			while( *(s + 2) >= *(s + 1) && *(s + 1) >= *s ){
				heron(s, p/2.0);
				*(s + 1)+=1;
				*(s + 2)-=1;
			}
	}
}

int main() {
	int t;
	cin>>t;
	while(t--){
		int p;
		cin>>p;
		::count = 0;
		sidesFromPerim(p);
		cout<<::count<<"\n";
		} 
	}
